viewing_party/party.py

```diff
@@ -1,23 +1,126 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    new_movie = {"title": title, "genre": genre, "rating": rating}
+    if title is None or genre is None or rating is None:
+        new_movie = None
+    return new_movie
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data 
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data 
+
+def watch_movie(user_data, title):
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            user_data["watched"].append(movie)
+            user_data["watchlist"].remove(movie)
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    number_of_movies = 0
+    average_ratings = 0.0
+    if len(user_data["watched"]) == 0:
+        average_ratings = 0.0
+        return average_ratings
+    for movie in user_data["watched"]:
+        number_of_movies += 1
+        average_ratings += float(movie["rating"])
+    average_ratings = average_ratings/number_of_movies
+    return average_ratings
+        
+def get_most_watched_genre(user_data):
+# to include multiple genres that have high count
+    genre_dict = {}
+    most_frequent_genre = ""
+    if len(user_data["watched"]) == 0:
+        return None
+    for movie in user_data["watched"]:
+        if movie["genre"] not in genre_dict:
+            genre_dict[movie["genre"]] = 1
+        elif movie["genre"] in genre_dict:
+            genre_dict[movie["genre"]] += 1
+    most_frequent_genre = [key for key, value in genre_dict.items()\
+         if value == max(genre_dict.values())]
+    return ' '.join(most_frequent_genre)
+
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
+def get_user_watched_movies(user_data):
+    user_watched_movies = []
+    for movie in user_data["watched"]:
+        user_watched_movies.append(movie)

```

Comment:
Recall that user_data["watched"] already returns a list of dictionaries that represent movies. Instead of iterating through `user_data["watched"]` and adding each `movie` into the new list `user_watched_movies`, whenever you'd like to access a user's watched movies, we can directly use `user_data["watched"]`